packages/maskbook/src/plugins/ITO/hooks/usePoolPayload.ts

```diff
@@ -1,6 +1,30 @@
-import { useAsyncRetry } from 'react-use'
+import { ValueRef } from '@dimensiondev/holoflows-kit'
 import { PluginITO_RPC } from '../messages'
+import { useValueRef } from '../../../utils/hooks/useValueRef'
+import type { JSON_PayloadInMask } from '../types'
+
+let storage = new Map<string, ValueRef<JSON_PayloadInMask>>()
+let storageOfLoading = new Map<string, boolean>()
 
 export function usePoolPayload(pid: string) {
-    return useAsyncRetry(() => PluginITO_RPC.getPool(pid), [pid])
+    if (!storageOfLoading.has(pid) || storageOfLoading.get(pid)) throw suspender(pid)
+    return {
+        payload: useValueRef(storage.get(pid)!),
+        retry: () => retry(pid),
+    }
+}
+
+async function retry(pid: string) {
+    storageOfLoading.set(pid, true)
```

Comment:
you can get rid of storageOfLoading, just `storage.delete(pid)` and trigger a new suspended